### PR TITLE
[CI] Add CI config for Native CPU

### PR DESCRIPTION
**Pull request by martygrant**

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -170,6 +170,7 @@ jobs:
           {name: HIP, triplet: amdgcn-amd-amdhsa, platform: ""},
           {name: L0, triplet: spir64, platform: ""},
           {name: OPENCL, triplet: spir64, platform: "Intel(R) OpenCL"}
+          {name: NATIVECPU, triplet: spir64, platform: ""}
```
Comment (lukaszstolarczuk) [inline .github/workflows/cmake.yml:173]: just want to confirm if you're aware - the `name` is also used as a label for self-hosted runner (see `runs-on: ${{matrix.adapter.name}}` below). I don't believe we have such a runner available at the moment.

Comment (lukaszstolarczuk) [reply .github/workflows/cmake.yml:173]: ok, the new runner should be up and running now - please verify if it works as needed 😉

Comment (martygrant) [reply .github/workflows/cmake.yml]: Thank you! I started a new job but realised the label for the runner is `NATIVECPU`, the naming in the workflow and CMake is `NATIVE_CPU`. Could you please edit the runner label? Thanks!

Comment (lukaszstolarczuk) [reply .github/workflows/cmake.yml]: Done

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -170,6 +170,7 @@ jobs:
           {name: HIP, platform: ""},
           {name: L0, platform: ""},
           {name: OPENCL, platform: "Intel(R) OpenCL"}
```
Comment (lukaszstolarczuk) [inline .github/workflows/cmake.yml:172]: you're missing a coma at the end of this line, and the workflow won't start

**Pull request merged**